src/NuGetGallery.Services/PackageManagement/PackageService.cs

```
@@ -698,9 +702,95 @@ private Package CreatePackageFromNuGetPackage(PackageRegistration packageRegistr
+                        // Only a package that is a tool package (and nothing else) will be matched against tools pattern set
+                        groups = items.FindItemGroups(conventions.Patterns.ToolsAssemblies);
+                    }
+                    else
+                    {
+                        var patterns = new[]
+                        {
+                            conventions.Patterns.CompileRefAssemblies,
+                            conventions.Patterns.CompileLibAssemblies,
+                            conventions.Patterns.RuntimeAssemblies,
+                            conventions.Patterns.ContentFiles,
+                            conventions.Patterns.ResourceAssemblies,
+                        };
+
+                        var msbuildPatterns = new[]
+                        {
+                            conventions.Patterns.MSBuildFiles,
+                            conventions.Patterns.MSBuildMultiTargetingFiles,
+                        };
+
+                        var standardGroups = patterns
+                            .SelectMany(p => items.FindItemGroups(p));
+
+                        // Filter out MSBuild assets that don't match the package ID.
+                        var packageId = nuspecReader.GetId();
+                        var msbuildGroups = msbuildPatterns
+                            .SelectMany(p => items.FindItemGroups(p))
+                            .Where(g => HasBuildItemsForPackageId(g.Items, packageId));
+
+                        groups = standardGroups.Concat(msbuildGroups);
+                    }
+
+                    supportedTFMs = groups
+                        .SelectMany(p => p.Properties)
+                        .Where(pair => pair.Key == ManagedCodeConventions.PropertyNames.TargetFrameworkMoniker)
+                        .Select(pair => pair.Value)
+                        .Cast<NuGetFramework>()
+                        .Distinct();
+                }
+                catch (Exception)
```

When would an exception be thrown? Should we log this? Consider explaining in the comment why the exception is ignored.

---

Agreed. If we can't analyze the assets of a package, it's quite likely that restore won't work on it either. And perhaps it should be rejected.

---

I've revised this logic--it had exception handling as a carry-over from when I was using it for package analysis and we had no nupkg for packages in test dbs. There's no need for it here.